### ISSUE TITLE
[Feature Request] Learn from selections and reprioritize suggestions

### ISSUE THREAD

**Issue by abogawat** [oliverschwendener/ueli#968]
Most similar apps I've used, including the Mac Spotlight, change the order of suggestions for a given query based on the user's selection. For example, if **Firefox** is the second choice when I type `fire`, but I manually select it a few times, it should start showing up as the top suggestion.

Ueli doesn't seem to do that. Am I missing a setting? If not, can this be added?

**Comment by oliverschwendener** [comment]
Ueli does a similar thing: the more you execute something, the higher it gets listed in the search results.

**Comment by abogawat** [comment]
That doesn't seem to be happening in my case. Case in point is the Firefox example I mentioned above. I've been using Ueli for over a month now, but Firefox shows up second in the list for the same query no matter how many times I've launched it so far. The same happens with a few other commands. Is there a setting for this that I might have unchecked by mistake?

**Comment by oliverschwendener** [comment]
No, there is no setting yet for this. This feature is always turned on. Apparently, the "algorithm" is too weak for your scenario. The optimal solution would be that a user could tweak the strength of this "algorithm" so that it could fit better to any setup. But as this is a bigger task, I can't promise you anything, sorry. 😅

**Comment by abogawat** [comment]
Got it. Thanks!

**Comment by tkohlmeier** [comment]
Am I missing something or does this feature now longer exist with version 9? If not, did you decide not to do that for some reason or did you just have no time to implement it? In that case, I would like to create a PR for this.

**Comment by oliverschwendener** [comment]
This feature was not ported from v8 to v9 as I find adding items that you use frequently to the favorites leads to much more accurate search results than the previous "learn from usage" implementation.

**Comment by tkohlmeier** [comment]
OK, so far that works good for me.
